include/anvill/Arch.h

```diff
@@ -0,0 +1,233 @@
+#pragma once
+
+#include <string>
+#include <vector>
+
+#include <llvm/IR/DebugInfoMetadata.h>
+#include <llvm/IR/Function.h>
+#include <llvm/IR/Instructions.h>
+#include <llvm/IR/IntrinsicInst.h>
+#include <llvm/IR/Metadata.h>
+#include <llvm/IR/Module.h>
+#include <llvm/IR/Type.h>
+
+#include "Decl.h"
+
+namespace remill {
+class Arch;
+class IntrinsicTable;
+struct Register;
+}  // namespace remill
+
+namespace anvill {
+
+enum SizeConstraint : unsigned {
+  kBit8 = (1 << 0),
+  kBit16 = (1 << 1),
+  kBit32 = (1 << 2),
+  kBit64 = (1 << 3),
+  kBit80 = (1 << 4),
+  kBit128 = (1 << 5),
+
+  kMaxBit128 = kBit128 | kBit80 | kBit64 | kBit32 | kBit16 | kBit8,
```

Comment:
I'm not sure about `kBit80` being in here. I don't believe we can pass `long double`s over `xmm` regs.

Reply:
I don't think this allows you to because we still have the type constraint. Since long double would be an integral type it wouldn't be able to be passed over xmm.